### PR TITLE
Add Middle East map

### PR DESCRIPTION
**Pull request by tjohnson314**
https://www.riograndegames.com/wp-content/uploads/2020/03/Power-Grid-Middle-East-Amp-South-Africa_rules2_2019.pdf

Summary of rule changes:
- Remove garbage and uranium plants from actual market during step 1
- Move to step 2 based on finishing the deck for the first time rather than the max number of cities built
- Custom total supply for each resource (Middle East has only 21 garbage and 11 uranium)
- Adjust oil prices when resupply is higher than available supply. There can be more than three oil available for $1, so this required changing the viewer as well.

### CODE REVIEW

File: engine/src/engine.ts [hunk 1]
```diff
@@ -278,6 +295,45 @@ export function setup(
...
+                });
+            } else {
+                G.powerPlantsDeck.push(plantToRemove);
+                G.log.push({
+                    type: 'event',
+                    event: `Sending Power Plant ${plantToRemove.number} to the bottom of the deck.`,
+                });
+            }
+
+            const newCurrentPlant: PowerPlant = G.futureMarket.shift()!;
```
Comment (LucasAMello) [inline engine/src/engine.ts]: What if the plant drawn from the deck belongs on the actual market?

File: engine/src/gamestate.ts [hunk 2]
```diff
@@ -110,6 +109,7 @@ export interface GameState {
     oilResupply?: number[][];
     garbageResupply?: number[][];
     uraniumResupply?: number[][];
+    oilPrices: number[]; // Adjust oil price for Middle East map.
```
Comment (LucasAMello) [inline engine/src/gamestate.ts]:
```suggestion
    oilPrices?: number[]; // Adjust oil price for Middle East map.
```

Comment (LucasAMello) [reply engine/src/gamestate.ts]: Remember that changes should be compatible with old games (in this case, old games won't have oilPrices).

File: viewer/src/components/boards/Resources.vue [hunk 3]
```diff
@@ -146,6 +146,21 @@
             <Coal :pieceId="-1" :targetState="{ x: 858, y: 57 }" :canClick="false" :transparent="true" :scale="0.2" />
         </template>
 
+        <template v-if="isMiddleEast">
```
Comment (LucasAMello) [inline viewer/src/components/boards/Resources.vue:173]: This is okay, but maybe consider changing it to how it is in USA recharged (add actual pieces of oil).

**Review by LucasAMello — CHANGES_REQUESTED** [review]

File: viewer/src/components/boards/Resources.vue (hunk 3 above)
Comment (tjohnson314) [reply viewer/src/components/boards/Resources.vue:173]: Thanks, I thought about that, but I'm not sure if there's enough space to add the pieces individually. I'd rather leave it like this for now.

File: engine/src/engine.ts (hunk 1 above)
Comment (tjohnson314) [reply engine/src/engine.ts]: Thanks, good catch! I've separated out the logic for this into the function removePlantsForMiddleEastStep1(). I had implemented it in two places in different ways, and this one was wrong.

File: engine/src/gamestate.ts (hunk 2 above)
Comment (tjohnson314) [reply engine/src/gamestate.ts]: Fixed.

**Review by LucasAMello — APPROVED** [review]

**Pull request merged**